iot/iot-device-registry-infinispan/src/main/java/io/enmasse/iot/registry/infinispan/credentials/CredentialsKey.java

```diff
@@ -3,20 +3,20 @@
  * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
  */
 
-package io.enmasse.iot.registry.infinispan;
+package io.enmasse.iot.registry.infinispan.credentials;
 
+import java.io.Serializable;
 import java.util.Objects;
-import org.infinispan.protostream.annotations.ProtoDoc;
-import org.infinispan.protostream.annotations.ProtoField;
 
 /**
  * A custom class to be used as key in the backend key-value storage.
  * This uses the uniques values of a credential to create a unique key to store the credentials details.
  *
  *  See {@link CacheCredentialService CacheCredentialService} class.
  */
-@ProtoDoc("@Indexed")
-public class CredentialsKey {
+public class CredentialsKey implements Serializable {
+
+    private static final long serialVersionUID = 1L;
 
     private String tenantId;
```

Comment:
Immutable objects are much easier to reason about, can we make these final to make this explicit?

Reply:
I am not sure, as it will be serialized by JBoss Marshalling.

Reply:
ok, they are effectively final anyway.